scripts/Muon/GUI/Common/muon_group.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import (absolute_import, division, print_function)
+
+from Muon.GUI.Common.muon_workspace import MuonWorkspace
+import six
+
+
+class MuonGroup(object):
+    """Simple structure to store information on a detector group.
+
+    The name is set at initialization and after that cannot be changed.
+    The detector list can be modified by passing a list of ints (type checks for this)
+    The number of detectors is stored
```

Comment:
Not possible, a new instance of the group would need to be created. This is a deliberate safety net.

Reply:
safety net for what?

Reply:
The name being changed. It is hard to be definitive without me actually referring to the code which uses this class. But suffice it to say it is deliberate; to protect against the name being changed because in the design I am implemented this is important.

Reply:
you might decide to change the name but keep the detectors, this would cause two almost identical objects. I think it would be more sensible to only allow valid names to be assigned when changing it (i.e. not an empty string)

Reply:
Yes you might, but that is exactly what I want to prevent happening so that my interface works. It must keep its name so that the workspace name which it also contains remains synchronised.

The code would become less stable as a result of making this change.

The "key" reason here (no pun intended) is that the name of the group is used as a key in a dictionary which stores groups. If the group name changes via `group.name = "new_name"` then the key no longer points to the correct group; then when it comes to handling groups in the muon_context all hell breaks loose.

Reply:
To add to this discussion : 

You can actually modify the name without instantiating a new object; you can use

```
group._name = "new_name"
```

However as suggested by the accessing of a "private" variable; this should not be done.